web/ui/dashboard/src/app/private/pages/account/personal-settings/personal-settings.component.html

```diff
@@ -112,5 +112,8 @@ <h3 class="font-semibold mb-28px">Generate Token</h3>
 	(closeModal)="showAccessKey = false; fetchPersonalKeys()"
 ></convoy-token-modal>
 
+
 <!-- revoke token modal  -->
-<app-delete-modal *ngIf="showRevokeKeyModal" [isLoading]="isRevokingKey" [deleteText]="'revoke “' + selectedKey?.name + 'api key”'" deleteButtonText="revoke" (closeModal)="showRevokeKeyModal = false" (deleteData)="revokeKey()"></app-delete-modal>
+<dialog #settingsDialog convoy-dialog position="center" size="sm">
+    <app-delete-modal [isLoading]="isRevokingKey" [deleteText]="'revoke “' + selectedKey?.name + 'api key”'" deleteButtonText="revoke" (closeModal)="settingsDialog.close()" (deleteData)="revokeKey()"></app-delete-modal>
+</dialog>
```

Comment:
I love the implementation here. This is exactly how our components and directives should work, working together without any hindrance, each element minding its own business and emitting events for anyone that cares to listen. 👍🏽